tidymodel-prediction-workflows-inside-snowflake-with-orbital/tidymodel-prediction-workflows-inside-snowflake-with-orbital.md

```diff
@@ -0,0 +1,1145 @@
+  step_mutate(TERM = (TERM == "60 months")) |>
+  step_mutate(across(!TERM, as.numeric)) |> 
+  step_normalize(all_numeric_predictors()) |>
+  step_impute_mean(all_of(c("BC_OPEN_TO_BUY", "BC_UTIL"))) |>   
+  step_filter(!if_any(everything(), is.na))
+```
+
+Our recipe `lendingclub_rec` defines our model formula: we'll use all the other variables in the dataset
+to predict the interest rate. We also perform a few pre-processing steps, including
+turning `TERM` into a logical, normalizing all numeric predictors, and removing
+missing values.
+
+Next, we define the type of model we want to fit: a linear model. `linear_reg()` 
+is from the [parsnip package](https://parsnip.tidymodels.org/).
+
+```r
+lendingclub_lr <- linear_reg()
+```
+
+Now, we can create our workflow, adding our linear model and pre-processing recipe. 
+
+```r
+lendingclub_wf <- 
+  workflow() |> 
+  add_model(lendingclub_lr) |> 
+  add_recipe(lendingclub_rec)
+```
+
+Our workflow object now looks like this:
+
+```r
+lendingclub_wf
+```
+
+```
+══ Workflow ═══════════════════════════════════════════════════════════════════════════════════════════════════════
+Preprocessor: Recipe
+Model: linear_reg()
+
+── Preprocessor ───────────────────────────────────────────────────────────────────────────────────────────────────
+5 Recipe Steps
+
+• step_mutate()
+• step_mutate()
+• step_normalize()
+• step_impute_mean()
+• step_filter()
+
+── Model ──────────────────────────────────────────────────────────────────────────────────────────────────────────
+Linear Regression Model Specification (regression)
+
+Computational engine: lm 
+```
+
+Notice that it includes our pre-processing steps and model specification. 
+
+### Fit model and compute metrics
+
+Now that we have our workflow object `lendingclub_wf`, we can use it to fit our model. 
+
+```r
+lendingclub_fit <- 
+  lendingclub_wf |> 
+  fit(data = lendingclub_prep)
+```
+
+To understand how well our model performs, we define a metric set using `yardstick::metric_set()`
+and then compute those metrics for `lendingclub_fit`.
+
+```r
+lendingclub_metric_set <- metric_set(rmse, mae, rsq)
+
+lendingclub_metrics <- 
+  lendingclub_fit |>
+  augment(lendingclub_prep) |>
+  lendingclub_metric_set(truth = INT_RATE, estimate = .pred)
+```
+
+`lendingclub_metrics` looks like this:
+
+```
+# A tibble: 3 × 3
+  .metric .estimator .estimate
+  <chr>   <chr>          <dbl>
+1 rmse    standard       4.38 
+2 mae     standard       3.43 
+3 rsq     standard       0.233
+```
+
+### Version model with vetiver
+
+The [vetiver](https://rstudio.github.io/vetiver-r/) package provides tools to version, share, 
+deploy, and monitor models. 
+
+We'll use vetiver to version and write our fitted model to a [pin](https://pins.rstudio.com/) on [Posit Connect](https://posit.co/products/enterprise/connect/). 
+This will allow us to identify which version of our model is active and track
+performance against other models over time. 
```

Comment:
We introduce Posit Connect here but access to Posit Connect isn't mentioned in the quickstart requirements.